src/service/topics.ts

```diff
@@ -127,6 +128,13 @@ const getLatestTopicPost = async (topicName: string): Promise<TopicLatestPostDat
       success: res?.data?.success
     };
   } catch (e) {
+    if (e?.response?.message === 'This topic has no active post')
+      return {
+        success: false,
+        message: e?.response?.message,
+        expired_at: moment().add(1, 'h').toISOString()
+      };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This error handling block is well implemented. It checks for a specific error message and returns a custom response object with an expiry time set one hour into the future using the `moment` library. This could be useful for caching or retry logic in the calling function.
```diff
+    if (e?.response?.message === 'This topic has no active post')
+      return {
+        success: false,
+        message: e?.response?.message,
+        expired_at: moment().add(1, 'h').toISOString()
+      };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->